shell/packages/sandstorm-db/db.js

```diff
@@ -1810,23 +1810,81 @@ if (Meteor.isServer) {
   // Cleanup tokens every hour.
   SandstormDb.periodicCleanup(3600000, cleanupExpiredAssetUploads);
 
-  SandstormDb.prototype.getIconUrls = function (icon) {
-    check(icon, Match.OneOf(undefined, null,
-                            { format: String, assetId: String,
-                              assetId2xDpi: Match.Optional(String), }));
-    const root =
-        Url.parse(process.env.ROOT_URL).protocol + "//" + this.makeWildcardHost("static") + "/";
+  SandstormDb.prototype.updateCachedViewInfo = function (grainId, viewInfo) {
```

Comment:
The refcounting happens when the grain specifically fills in `ViewInfo.metadata`. I wouldn't want to rule out the possibility that this could become the usual case in the future, e.g. if it becomes common for apps to provide multiple types of actions and to present a different grain icon for each action.